reports/downloadDashboardData.ts

```diff
@@ -0,0 +1,66 @@
+import { appIds, daysOfDataInMs } from './constants.js'
+import { doCountlyFetch } from './doCountlyFetch.js'
+import { writeFile } from 'node:fs/promises'
+
+interface ActiveUsersResponse {
+  calculating: boolean
+  data: Record<string, {
+    d: number
+    w: number
+    m: number
+  }>
+}
+
+/**
+ * Print out a CSV of the number of unique users per day for each app
+ */
+export async function downloadDashboardData (): Promise<void> {
+  let printedHeaders = false
+  const dailyArray = []
+  const weeklyArray = []
+  const monthlyArray = []
+  for (const [appName, appId] of Object.entries(appIds)) {
+    // let calculating = true
+    let response: ActiveUsersResponse = { calculating: true, data: {} }
+    while (response.calculating) {
+      /**
+       * @see https://api.count.ly/reference/oanalyticssessions
+       */
+      response = await doCountlyFetch({ path: '/o/active_users', appId, extraParams: `period=[${new Date().getTime() - daysOfDataInMs}, ${new Date().getTime()}]` })
+      // eslint-disable-next-line no-console
+      console.log(`${appName} calculating? `, response.calculating)
+      if (response.calculating) {
+        await new Promise((resolve) => setTimeout(resolve, 4000))
+      }
+    }
+    const activeUserData = []
+    for (const [key, value] of Object.entries(response.data)) {
+      activeUserData.push({
+        _id: key,
+        ...value
+      })
+    }
```

Comment:
We totally could. The runtime is extremely fast when we're not waiting for countly `calculating` though, so I wasn't concerned. Also trying to keep requests sane for our countly server (not sure what load it can take)

Will update, so if we do wait for calculating, we will move forward on the next req.

Reply:
Actually, if its in parallel, order will get jacked up. Thats the other reason, so i would need to enforce ordering some other way, and I've already got charts dependent upon `G7:g13` in the google spreadsheet. 

Probably would be good to update the spreadsheet to select column for a certain "app name" to harden it, but it might be easier to just leave this alone. I'll sleep on it

Reply:
The rows are not being sorted after, the columns are. The row sorting is what needs to remain in place to not bork the google spreadsheet formulas.

Reply:
done.. now we have an additional loop over appIds in order to ensure the correct ordering.